### PR TITLE
Align the `-server.http-write-timeout` with `-querier.timeout` to 2m

### PR DESCRIPTION
**Pull request by hi-rustin**
#### What this PR does
Align the `-server.http-write-timeout` with `-querier.timeout` to 2m. Add validation for these values.
#### Which issue(s) this PR fixes or relates to

Fixes https://github.com/grafana/mimir/issues/3248

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/querier/querier.go [hunk 1]
```diff
@@ -79,14 +79,18 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
...
+func (cfg *Config) Validate(httpServerWriteTimeout time.Duration) error {
 	// Ensure the config wont create a situation where no queriers are returned.
 	if cfg.QueryIngestersWithin != 0 && cfg.QueryStoreAfter != 0 {
 		if cfg.QueryStoreAfter >= cfg.QueryIngestersWithin {
 			return errBadLookbackConfigs
 		}
 	}
 
+	if cfg.EngineConfig.Timeout > httpServerWriteTimeout {
+		return fmt.Errorf("querier timeout (%s) must be lower than HTTP server write timeout (%s)", cfg.EngineConfig.Timeout, httpServerWriteTimeout)
```
Comment (dimitarvdimitrov) [inline pkg/querier/querier.go:91]: What do you think about moving this into the `pkg/mimir` package? Along with the [other validations](https://github.com/grafana/mimir/blob/cca6e117cc537e4b68f5d31e27ca7c7eeda43618/pkg/mimir/mimir.go#L207-L212) that rely on multiple configuration structs?

I think that way we'll have easier testing and arguably maybe better separation of concerns in the querier package.

**Review by dimitarvdimitrov — COMMENTED** [review]
Thanks for picking this up :) I think we can move it into a higher level package

File: pkg/querier/querier.go (hunk 1 above)
Comment (hi-rustin) [reply pkg/querier/querier.go]: Godd sugesstion! Thanks!

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM <3

I'll let others have a look before merging

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,6 +20,7 @@
 * [ENHANCEMENT] Distributor: Include `X-Scope-OrgId` header in requests forwarded to configured forwarding endpoint. #3283
 * [ENHANCEMENT] Alertmanager: reduced memory utilization in Mimir clusters with a large number of tenants. #3309
 * [ENHANCEMENT] Add experimental flag `-shutdown-delay` to allow components to wait after receiving SIGTERM and before stopping. In this time the component returns 503 from /ready endpoint. #3298
+* [ENHANCEMENT] Align the `-server.http-write-timeout` with `-querier.timeout` to 2m. #3346
```
Comment (pracucci) [inline CHANGELOG.md:23]: We use CHANGE for any change to default config. Please move it above among other CHANGE entries.

```suggestion
* [CHANGE] The default value of `-server.http-write-timeout` has changed from 30s to 2m. #3346
```

File: pkg/mimir/mimir.go [hunk 3]
```diff
@@ -225,6 +225,10 @@ func (c *Config) Validate(log log.Logger) error {
 	if err := c.Querier.Validate(); err != nil {
 		return errors.Wrap(err, "invalid querier config")
 	}
+	if c.Querier.EngineConfig.Timeout > c.Server.HTTPServerWriteTimeout {
+		return fmt.Errorf("querier timeout (%s) must be lower than HTTP server write timeout (%s)",
```
Comment (pracucci) [inline pkg/mimir/mimir.go:229]:
```suggestion
		return fmt.Errorf("querier timeout (%s) must be lower than or equal to HTTP server write timeout (%s)",
```

**Review by pracucci — APPROVED** [review]
LGTM, thanks! I left a couple of nits I'm going to auto apply.

File: pkg/mimir/mimir.go (hunk 3 above)
Comment (hi-rustin) [reply pkg/mimir/mimir.go]: Good catch! 
Thanks for your review!  💚 💙 💜 💛 ❤️

**Pull request merged**